compiler/rustc_interface/src/queries.rs

```diff
@@ -171,6 +172,9 @@ impl<'tcx> Queries<'tcx> {
                 )));
                 feed.crate_for_resolver(tcx.arena.alloc(Steal::new((krate, pre_configured_attrs))));
                 feed.output_filenames(Arc::new(outputs));
+
+                let feed = tcx.feed_local_def_id(CRATE_DEF_ID);
+                feed.def_kind(DefKind::Mod);
```

Comment:
I have a question regarding incremental builds: when referring to queries such as the results of ﻿`tcx.def_kind(LocalDefId)` or `﻿tcx.def_kind(DefId)`, does this mean the results are retrieved directly here?

Reply:
Here the result is put directly into the query cache.
So when `tcx.def_kind(def_id)` is called it will look into the query cache first, and will always find the result there, and will never call the query implementation (which now would panic because `providers.def_kind` is not set).